framework-api-vulnerabilities/activities/clean-up/deploy/Makefile

```diff
@@ -0,0 +1,20 @@
+FILE	:= ../flag
+FLAG	:= $(shell cat $(FILE))
+
+run: generate build
+	docker run -d -p 7004:80 --name sss-web-07_clean-up sss-web-07_clean-up
+
+build: generate
+	docker build -f Dockerfile -t sss-web-07_clean-up ..
+	rm ../src/data/users/user-information-expanded.json
+
+generate:
+	sed 's/__TEMPLATE__/$(FLAG)/g' ../src/data/users/user-information-expanded-template.json > ../src/data/users/user-information-expanded.json
+
+stop:
+	docker stop sss-web-07_clean-up
+
+clean: stop
+	docker rm sss-web-07_clean-up
+
+.PHONY: build run stop clean
```

Comment:
Since this file is always 90% the same with the others my proposal is to use a general Makefile.

It should be something like:
```
INTERNAL_PORT := 80
CONT_NAME := ${IMG_NAME}

build:
	docker build -t $(IMG_NAME) -f Dockerfile ..

run: build
	docker run -d -p $(EXTERNAL_PORT):$(INTERNAL_PORT) --name $(CONT_NAME) -t $(IMG_NAME)

stop:
	docker stop $(CONT_NAME)

clean: stop
	docker rm $(IMG_NAME)
	docker image rm $(IMG_NAME):latest

.PHONY: build run stop clean
```

It should be placed in the root of the repo in the `common` folder, something like `challenge.mk`.
Then in this Makefile we should only import and assign the variables:

```
EXTERNAL_PORT := 8001
IMG_NAME := sss-essentials-04_eyes

include ../../../../common/challenge.mk
```

This is an example from the `essentials` repo.
What do you think?

Reply:
Like the idea! But do you want to do it in this PR? We need to also update the other session folders.

Reply:
You're right.
Let's take small steps, I will open an issue related to this and we will deal with it in the future.
I will mark it as solved.